java-spiffe-core/src/main/java/io/spiffe/workloadapi/Address.java

```diff
@@ -0,0 +1,153 @@
+package io.spiffe.workloadapi;
+
+import io.spiffe.exception.SocketEndpointAddressException;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Utility class to get the default Workload API address and parse string addresses.
+ */
+public class Address {
+
+    /**
+     * Environment variable holding the default Workload API address.
+     */
+    public static final String SOCKET_ENV_VARIABLE = "SPIFFE_ENDPOINT_SOCKET";
+
+    private static final String UNIX_SCHEME = "unix";
+    private static final String TCP_SCHEME = "tcp";
+    private static final List<String> VALID_SCHEMES = Arrays.asList(UNIX_SCHEME, TCP_SCHEME);
```

Comment:
nit: probably better to represent as a `Set<String>` to keep the `.contains()` operation as O(1).

Reply:
Done.